ui/spells/spells.scss

```diff
@@ -125,6 +125,116 @@
     /* and make rows overlap */
     margin-top: -1px;
   }
+
+  $grid-padding: 2px;
+  $grid-column-gap: 2px;
+
+  @mixin spells-header($grid-template) {
+    display: grid;
+    grid-template-columns: $grid-template;
+    column-gap: $grid-column-gap;
+    background: #93c6f2;
+    padding: $grid-padding;
+
+    > div {
+      text-align: center;
+      margin: auto;
+      font-weight: bold;
+      width: 100%;
+    }
+  }
+
+  @mixin spells-rows($grid-template, $column-count) {
+    display: grid;
+    grid-template-columns: $grid-template;
+    row-gap: 0;
+    column-gap: $grid-column-gap;
+    background: aliceblue;
+    input, input[type=number], textarea, select {
+      width: 100%;
+      margin-left: unset;
+      background-color: white;
+    }
+    input[type=text] {
+      text-align: left;
+    }
+    input, span, select {
+      white-space: nowrap;
+    }
+    span {
+      text-align: center;
+      margin: auto 0;
+    }
+    button, button[type=roll] {
+      margin: 0;
+    }
+    .spell-row.conditional-one {
+      // This makes the notes span all columns
+      grid-column-start: 1;
+      grid-column-end: ($column-count + 1);
+    }
+  }
+
+  @mixin spells-grid($spell-type, $grid-template, $column-count) {
+    .spells-header-grid.#{$spell-type} {
+      @include spells-header($grid-template);
+    }
+
+    div[data-groupname="repeating_#{$spell-type}"] > div {
+      @include spells-rows($grid-template, $column-count);
+    }
+  }
+
+  $arcane-grid-template:
+    // Discipline?
+    minmax(0, 1fr)
+    // Name
+    minmax(0, 4fr)
+    // Ability
+    minmax(0, 0.75fr)
+    // Mod
+    minmax(0, 1fr)
+    // (Dice)
+    minmax(0, 0.5fr)
+    // EP
+    minmax(0, 1fr)
+    // Cast Time
+    minmax(0, 1fr)
+    // Touch/ Ray/ Missile
+    minmax(0, 2fr)
+    // Range
+    minmax(0, 1fr)
+    // Target
+    minmax(0, 1fr)
+    // Damage/ Duration
+    minmax(0, 1.5fr)
+    // N(otes)
+    minmax(0, 0.5fr)
+    // TP
+    minmax(0, 0.5fr)
+    // CP
+    minmax(0, 0.5fr)
+    // Exper.
+    minmax(0, 1fr)
+    // IQ ±
+    minmax(0, 0.5fr)
+    // Base
+    minmax(0, 1fr);
+
+  @include spells-grid('arcane', $arcane-grid-template, 18);
```

Comment:
By passing three variables, we can keep things consistent between the spell types.